### PR TITLE
#161779869 CRUD operations for commenting on an article

### PR DESCRIPTION
**Pull request by V-84**
#### What does this PR do?
Performs crud operations for commenting on articles
#### Description of Task to be completed?
- users can post comments on an article
- users can get all comments from an article
- users get only one comment for an article
- users can only update comment they created
- comments can only be deleted by the superAdmin and owner of the comment
#### How should this be manually tested?
pull branch
run npm run start

#### Any background context you want to provide?
n/a
#### What are the relevant pivotal tracker stories?
#161779869 
#### Screenshots (if appropriate)
n/a
#### Questions:
n/a

### CODE REVIEW

File: server/models/comment.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 /**
-* @param {object} sequelize
-* @param {object} DataTypes
-* @returns {object} Comment model
-*/
+ * @param {object} sequelize
+ * @param {object} DataTypes
+ * @returns {object} Comment model
+ */
```
Comment (VictoriaOffoma) [inline server/models/comment.js:5]: Lets make our indentation consistent for readability reason.

File: server/controllers/commentController.js [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+   * @description only owners of comment should be able to update
+   * @param {object} req - req from route
+   * @param {object} res - respose to route
+   * @param {object} next - callback function
+   * @returns {object} a response object
+   */
+  static async updateComment(req, res, next) {
+    const { user:{ id },body:{ comment }, params:{ commentId }} = req;
+    try {
+      const checkCommentOwner = await Comment.findOne({
```
Comment (VictoriaOffoma) [inline server/controllers/commentController.js]: Change "checkCommentOwner" to more suitable variable name.

File: server/controllers/commentController.js [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+          userId: id,
+         }
+      });
+      if (!checkCommentOwner) {
+        return res.status(403).json({
+          success: false,
+          message: 'User cannot update comment they did\'t create',
+        });
+      }
+      await Comment.update(
```
Comment (VictoriaOffoma) [inline server/controllers/commentController.js:135]: Lets write this in a more concise form.

File: server/controllers/commentController.js [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+    }
+  }
+
+   /**
+   * @description A function that deletes comments from the database
+   * @param {uuid} commentId - req from route
+   * @param {uuid} userId - respose to route
+   * @returns {boolean} a response object
+   */
+  static async deleteCommentFunction(commentId,userId){
```
Comment (VictoriaOffoma) [inline server/controllers/commentController.js]: Remove the function in "deleteCommentFunction"

File: server/controllers/commentController.js [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+    const comment = await Comment.findOne({
+      where: {
+        id: commentId,
+        userId,
+      }
+    });
+    if (!comment) {
+      return false;
+
```
Comment (VictoriaOffoma) [inline server/controllers/commentController.js]: remove space.

File: server/controllers/commentController.js [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+  }
+
+  /**
+   * @description Users should be able to delete only their comment
+   * @param {object} req - req from route
+   * @param {object} res - respose to route
+   * @param {object} next - callback function
+   * @returns {object} a response object
+   */
+  static async deleteComment(req, res, next) {
```
Comment (VictoriaOffoma) [inline server/controllers/commentController.js:172]: You do not need two functions for delete operation. One function should handle ascertaining who is able to delete a comment while the other should handle the delete process.

File: server/helpers/checkValidArticle.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
+import db from '../models';
+
+const { Article } = db;
+
+/**
+ * @description checks is an article exists in the database
+ * @param {string} id - req from route
+ * @returns {boolean} - depending on if ID is valid
+ */
+const checkArticleId = async (id) => {
```
Comment (VictoriaOffoma) [inline server/helpers/checkValidArticle.js:10]: Do we not have something that checks the validity of articles already?

File: server/helpers/checkValidCommentId.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
+import db from '../models';
+
+const { Comment} = db;
+
+/**
+ * @description checks is an comment exists in the database
+ * @param {string} id - req from route
+ * @returns {boolean} - depending on if ID is valid
+ */
+const checkCommentId = async (id) => {
```
Comment (VictoriaOffoma) [inline server/helpers/checkValidCommentId.js:10]: The file name and the function name is not a true reflection of what the function does. Do change them to something more fitting.

File: server/helpers/validationHelper.js [hunk 9]
```diff
@@ -51,11 +51,25 @@ const  isSocialMediaEmail = (email) => {
   return email.includes('@facebook.com') || email.includes('@twitter.com');
 };
 
+
+/**
+ * @description Validates a text field
+ * @param {String} text
+ * @return {Boolean} returns true or false
+ */
+const validateText = (text) => {
```
Comment (VictoriaOffoma) [inline server/helpers/validationHelper.js]: What text are we validating here?

File: server/models/comment.js [hunk 10]
```diff
@@ -1,8 +1,8 @@
...
 export default (sequelize, DataTypes) => {
   const Comment = sequelize.define('Comment', {
     id: {
```
Comment (VictoriaOffoma) [inline server/models/comment.js:8]: Having the model and a property in the model sharing the same name will create confusion. Alternative names for that property in question could be "content", "body", "message".

File: server/controllers/commentController.js (hunk 2 above)
Comment (V-84) [reply server/controllers/commentController.js]: ok

File: server/controllers/commentController.js (hunk 3 above)
Comment (V-84) [reply server/controllers/commentController.js:135]: updated to `User is not authorized to update comment`

File: server/controllers/commentController.js (hunk 4 above)
Comment (V-84) [reply server/controllers/commentController.js]: done

File: server/controllers/commentController.js (hunk 5 above)
Comment (V-84) [reply server/controllers/commentController.js]: done

File: server/controllers/commentController.js (hunk 6 above)
Comment (V-84) [reply server/controllers/commentController.js:172]: @VictoriaOffoma Please I need more clarity on this

File: server/controllers/commentController.js [hunk 11]
```diff
@@ -0,0 +1,202 @@
...
+  }
+
+  /**
+   * @description Handles who can delete comment
+   * @param {object} req - req from route
+   * @param {object} res - respose to route
+   * @param {object} next - callback function
+   * @returns {object} a response object
+   */
+  static async deleteComment(req, res, next) {
```
Comment (VictoriaOffoma) [inline server/controllers/commentController.js:172]: This was what i was talking about: https://gist.github.com/VictoriaOffoma/549cc03934befb7d7fd4fbcf11189988

**Review by VictoriaOffoma — APPROVED** [review]

**Pull request merged**